### PR TITLE
fix(summarize): Use dict keys as metric names

### LINKED ISSUE #3249: Dict-form custom metrics are not distinguishable
When computing custom metrics which output a dict, `report.metrics.summarize()` outputs ambiguous-looking metric names:

```python
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression

from skore import evaluate

X, y = make_classification(return_X_y=True)

estimator_report = evaluate(
    LogisticRegression(),
    X,
    y,
    pos_label=1,
)

def custom(estimator, X, y):
    return {"A": 1, "B": 2}

estimator_report.metrics.add(custom)

# "custom" appears twice (one for "A", one for "B")
estimator_report.metrics.summarize().frame()
# metric
# custom          1.000000
# custom          2.000000
# accuracy        1.000000
# precision       1.000000
# recall          1.000000
# roc_auc         1.000000
# log_loss        0.100168
# brier_score     0.022797
# fit_time        0.004651
# predict_time    0.000376
# Name: LogisticRegression, dtype: float64

# Here "A" and "B" appear correctly
estimator_report.metrics.summarize().frame(flat_index=False, verbose_name=True)
# Metric
# A                   1.000000
# B                   2.000000
# Accuracy            1.000000
# Precision           1.000000
# Recall              1.000000
# ROC AUC             1.000000
# Log loss            0.100168
# Brier score         0.022797
# Fit time (s)        0.004651
# Predict time (s)    0.000376
# Name: LogisticRegression, dtype: float64
```

I would expect "A" and "B" to appear in the first output as well.

### PR DESCRIPTION
**Pull request by auguste-probabl**
Closes https://github.com/probabl-ai/skore/issues/3249

```py
from sklearn.datasets import make_classification
from sklearn.linear_model import LogisticRegression

from skore import evaluate

X, y = make_classification(return_X_y=True)

estimator_report = evaluate(
    LogisticRegression(),
    X,
    y,
    pos_label=1,
)

def custom(estimator, X, y):
    return {"A": 1, "B": 2}

estimator_report.metrics.add(custom)

# "A" and "B" are used intead of "custom"
estimator_report.metrics.summarize().frame()
# metric
# A               1.000000
# B               2.000000
# accuracy        0.800000
# precision       0.888889
# recall          0.727273
# roc_auc         0.969697
# log_loss        0.304507
# brier_score     0.100096
# fit_time        0.005184
# predict_time    0.000279
# Name: LogisticRegression, dtype: float64

# "A" and "B" appear alongside verbose-named metrics
estimator_report.metrics.summarize().frame(flat_index=False, verbose_name=True)
# Metric
# A                   1.000000
# B                   2.000000
# Accuracy            1.000000
# Precision           1.000000
# Recall              1.000000
# ROC AUC             1.000000
# Log loss            0.064413
# Brier score         0.013533
# Fit time (s)        0.002224
# Predict time (s)    0.000121
# Name: LogisticRegression, dtype: float64
```

### CODE REVIEW

**Review by jeromedockes — APPROVED** [review]
LGTM, thanks a lot @auguste-probabl
